### PR TITLE
🗂 Update paper_index section

### PR DESCRIPTION
**Pull request by behroozazarkhalili**
## Summary
This PR updates the paper index documentation to include additional research papers and their corresponding implementation configurations within the TRL package.

## Changes
- Added new paper entries with proper references and configuration examples
- Included implementation details and parameter settings for reproducibility
- Enhanced documentation to help users understand and reproduce training configurations

This update expands the paper index to provide better coverage of research implementations available in TRL, making it easier for users to find and use relevant configurations for their research and experiments.

### CODE REVIEW

**Comment by qgallouedec** [comment]
Can you also mention that we don't support dynamic sampling

**Comment by behroozazarkhalili** [comment]
> Can you also mention that we don't support dynamic sampling

Done. I also added `mask_truncated_completions = True` to the configuration.

File: docs/source/paper_index.md [hunk 1]
```diff
@@ -26,6 +26,58 @@ training_args = GRPOConfig(
...
+**📜 Paper**: https://huggingface.co/papers/2503.20783
+
+A study of R1-Zero training identifies pretraining effects on RL performance and proffers Dr. GRPO to enhance token efficiency, achieving superior accuracy on AIME 2024. To reproduce the paper's setting, use this configuration:
+
+```python
+from trl import GRPOConfig
+
+training_args = GRPOConfig(
+    loss_type="dr_grpo",
+    per_device_train_batch_size=1, # train_batch_size_per_device in the Training section of the repository
```
Comment (qgallouedec) [inline docs/source/paper_index.md:81]: these comments are very good practice IMO!

File: docs/source/paper_index.md [hunk 2]
```diff
@@ -26,6 +26,66 @@ training_args = GRPOConfig(
...
+    beta=0.0 # section 2.3, DAPO paper
+
+)
+```
+
+To add Soft Overlong Punishment, add the following to the `reward_funcs` on the `GRPOTrainer`:
+
+```python
+sop_reward = get_soft_overlong_punishment(max_completion_len=20480, soft_punish_cache=4096) # DAPO paper: section 4.1
+
```
Comment (qgallouedec) [inline docs/source/paper_index.md]:
```suggestion
# Soft Overlong Punishment
sop_reward = get_soft_overlong_punishment(max_completion_len=20480, soft_punish_cache=4096) # DAPO paper: section 4.1
trainer = GRPOTrainer(
    ...,
    args=training_args,
    reward_funcs=[..., sop_reward],
)
```

File: docs/source/paper_index.md [hunk 3]
```diff
@@ -26,6 +26,66 @@ training_args = GRPOConfig(
...
+- Overlong Filtering
+- Clip-Higher
+- Soft Overlong Punishment
+- Token-level Loss
+- Dynamic Sampling (⚠️ Not supported in TRL)
+
+To reproduce the paper's setting, use this configuration:
+
+```python
+from trl import GRPOConfig
```
Comment (qgallouedec) [inline docs/source/paper_index.md]:
```suggestion
from trl import GRPOConfig, GRPOTrainer
```
